aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -642,44 +628,128 @@ private CreateSqlCommands getCreateCommands(@NonNull Set<String> modelNames) {
-        final SQLiteStatement compiledSqlStatement = sqlCommand.getCompiledSqlStatement();
-        // 1-based index for columns
-        int columnIndex = 1;
+    // extract model field values to save in database
+    private List<Object> extractFieldValues(@NonNull Model model) throws DataStoreException {
+        final String modelName = getModelName(model);
+        final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
+        final SQLiteTable table = SQLiteTable.fromSchema(schema);
+        final SQLiteModelFieldTypeConverter converter =
+                new SQLiteModelFieldTypeConverter(schema, modelSchemaRegistry, gson);
+        final Map<String, ModelField> modelFields = schema.getFields();
+        final List<Object> bindings = new ArrayList<>();
+        for (SQLiteColumn column : table.getSortedColumns()) {
+            final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
+            final Object fieldValue = converter.convertValueFromTarget(model, modelField);
+            bindings.add(fieldValue);
+        }
+        return bindings;
+    }
 
-        // bind model field values to sql columns
-        if (model != null) {
-            final String modelName = getModelName(model);
-            final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
-            final SQLiteModelFieldTypeConverter converter =
-                    new SQLiteModelFieldTypeConverter(schema, modelSchemaRegistry, gson);
-            final Map<String, ModelField> modelFields = schema.getFields();
-
-            final List<SQLiteColumn> columns = sqlCommand.getColumns();
-
-            for (SQLiteColumn column : columns) {
-                final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
-                final Object fieldValue = converter.convertValueFromTarget(model, modelField);
-                bindValueToStatement(compiledSqlStatement, columnIndex, fieldValue);
-                columnIndex++;
+    private List<Model> cascade(Collection<? extends Model> models) {
+        // Use sqliteModelTree to identify every model affected by cascading delete.
+        Map<ModelSchema, Set<String>> descendants =
+                sqliteModelTree.descendantsOf(models);
+
+        List<Model> cascadedModels = new ArrayList<>();
+        for (Map.Entry<ModelSchema, Set<String>> entry : descendants.entrySet()) {
+            ModelSchema schema = entry.getKey();
+            for (String id : entry.getValue()) {
+                // Create dummy model instance using just the ID and model type
+                String dummyJson = gson.toJson(Collections.singletonMap("id", id));
+                Model dummyItem = gson.fromJson(dummyJson, schema.getModelClass());
+                cascadedModels.add(dummyItem);
             }
         }
+        return cascadedModels;
+    }
+
+    private <T extends Model> void writeData(
```

Comment:
This is a nice method to help clean things up a bit!

Reply:
yep! previously, there were a lot of copy-pasted code throughout for working with database commands, so I wanted to consolidate it into this single method for readability and cleanliness.